app/platform/__init__.py

```diff
@@ -373,3 +378,154 @@ def edit_form(grant_id: UUID4, collection_id: UUID4, section_id: UUID4, form_id:
+                question_data_type=question_data_type,
+            )
+        )
+    return render_template(
+        "platform/developers/choose_question_type.html",
+        grant=db_form.section.collection_schema.grant,
+        collection=db_form.section.collection_schema,
+        section=db_form.section,
+        db_form=db_form,
+        form=wt_form,
+    )
+
+
+@platform_blueprint.route(
+    "/grants/<uuid:grant_id>/developers/collections/<uuid:collection_id>/sections/<uuid:section_id>/forms/<uuid:form_id>/questions/add",
+    methods=["GET", "POST"],
+)
+@mhclg_login_required
+@auto_commit_after_request
+def add_question(grant_id: UUID4, collection_id: UUID4, section_id: UUID4, form_id: UUID4) -> ResponseReturnValue:
+    form = get_form_by_id(form_id)
+    question_data_type_arg = request.args.get("question_data_type", None)
+    question_data_type_enum = QuestionDataType.coerce(question_data_type_arg)
+
+    wt_form = QuestionForm()
+    if wt_form.validate_on_submit():
+        try:
+            assert wt_form.text.data is not None
+            assert wt_form.hint.data is not None
+            assert wt_form.name.data is not None
+            create_question(
+                form=form,
+                text=wt_form.text.data,
+                hint=wt_form.hint.data,
+                name=wt_form.name.data,
+                data_type=question_data_type_enum,
+            )
+            return redirect(
+                url_for(
+                    "platform.manage_form",
+                    grant_id=grant_id,
+                    collection_id=collection_id,
+                    section_id=section_id,
+                    form_id=form_id,
+                    back_link="manage_section",
+                )
+            )
+        except DuplicateValueError as e:
+            field_with_error: Field = getattr(wt_form, e.field_name)
+            field_with_error.errors.append(f"{field_with_error.name.capitalize()} already in use")  # type:ignore[attr-defined]
+
+    return render_template(
+        "platform/developers/add_question.html",
+        grant=form.section.collection_schema.grant,
+        collection=form.section.collection_schema,
+        section=form.section,
+        db_form=form,
+        chosen_question_data_type=question_data_type_enum,
+        form=wt_form,
+    )
+
+
+@platform_blueprint.route(
+    "/grants/<uuid:grant_id>/developers/collections/<uuid:collection_id>/sections/<uuid:section_id>/forms/<uuid:form_id>/questions/<uuid:question_id>/move/<string:direction>",
+    methods=["POST"],
+)
+@mhclg_login_required
+@auto_commit_after_request
+def move_question(
+    grant_id: UUID4, collection_id: UUID4, section_id: UUID4, form_id: UUID4, question_id: UUID4, direction: str
+) -> ResponseReturnValue:
+    question = get_question_by_id(question_id=question_id)
+
+    if direction == "up":
+        move_question_up(question)
+    elif direction == "down":
+        move_question_down(question)
```

Comment:
```suggestion
    if direction == "up":
        move_question_up(question)
    elif direction == "down":
        move_question_down(question)
    else:
        abort(400)
```

Might help us catch if we ever mess up the thing we pass into `direction`